src/Router/Route.php

```diff
@@ -133,6 +151,14 @@ public function setOptions(array $options)
     }
 
     /**
+     * @param string $name
+     */
+    public function setName($name)
```

Comment:
@weierophinney Are you sure about that? I can create a Route without a name in constructor, setName() after that and use it to add a route to a Route Adapter. This is perfectly fine. The name is auto-generated if you don't provide it in the constructor but you can change it with setName() if you want.